### ISSUE TITLE
Clarify whether Float8Linear is expected to support pure E5M2 training

### ISSUE THREAD

**Issue by wuyufffan** [AMD-AGI/Primus-Turbo#359]
Hi Turbo team,

I noticed that `tests/pytorch/modules/test_linear_fp8.py::test_float8linear`
currently parametrizes:

```python
@pytest.mark.parametrize("format", [Format.E4M3, Format.E5M2])
```

For the tensorwise `Format.E5M2` case, the current `gemm_fp8.py` dtype policy appears to quantize all training GEMMs as pure E5M2:

- fprop: input E5M2, weight E5M2
- dgrad: weight E5M2, gradient E5M2
- wgrad: input E5M2, gradient E5M2

This seems different from NVIDIA TransformerEngine's FP8 training policy. In TE recipes, pure E5M2 training is explicitly rejected:

```python
assert self.fp8_format != Format.E5M2, "Pure E5M2 training is not supported."
```

TE's HYBRID behavior is instead:

- fprop: input E4M3, weight E4M3
- dgrad: weight E4M3, gradient E5M2
- wgrad: input E4M3, gradient E5M2

Could you clarify the intended semantics in Turbo?

1. Is `Float8Linear(..., Format.E5M2)` intended to support pure E5M2 training?
2. If yes, what accuracy expectation should this test enforce, given TE does not support pure E5M2 training?
3. If no, should Turbo align `Format.E5M2` training with HYBRID semantics, or should the pure E5M2 training test be removed/skipped/xfail?
4. Was the E5M2 `Float8Linear` test added as intentional training coverage in PR #145, or as broad API coverage during the `Float8Linear` refactor?

Context:
- PR #145 added `Float8Linear` and parametrized `format=[Format.E4M3, Format.E5M2]`.
- Earlier PR #36 had removed E5M2 from the old `MXLinear` test.
- TE explicitly disallows pure E5M2 training.

Thanks.

**Comment by RuibinCheung** [comment]
Thanks for your feedback. 
1. For FP8 GEMM kernel we hope support general format (etc. E4M3, E5M2 and HYBRID). But as your said the E5M2 training is not applied in popular framework currently.
2. I think the `Float8Linear` is designed to support torchtitan initially. So we forgot to add HYBRID support. I will submit a PR to fix this issue.

**Comment by wuyufffan** [comment]
Thanks for the clarification.

Could you also help confirm the kernel/test coverage side?

1. On AMD platforms such as `gfx942` and `gfx950`, does the current pure `E5M2 x E5M2` path pass the existing FP8 GEMM / `Float8Linear` unit tests in your CI or local validation?

2. For pure `Format.E4M3` and pure `Format.E5M2`, are they tested with exactly the same quantization strategy, with only the target FP8 dtype changed?

More specifically, are scale computation, amax handling, scaling granularity, reference path, and SNR/tolerance policy intended to be identical between E4M3 and E5M2, except for the FP8 format/range itself?

I am asking because in our local validation environment, pure E5M2 does not pass when using the same quantization policy, while the current parameterized tests seem to imply that pure E5M2 is expected to be testable. I want to distinguish whether this is:

- an unsupported training semantic that should become HYBRID,
- a platform/kernel coverage gap,
- or an expected accuracy difference that needs a different test policy for E5M2.

Thanks.

**Comment by RuibinCheung** [comment]
1. I have submitted a PR(https://github.com/AMD-AGI/Primus-Turbo/pull/360) to add HYBRID case for `Float8Linear`.
2. For pure E5M2 or HYBRID the quantization strategy is consistent in tensorwise granularity but there are some different in mxfp8 granularity. I think you need to attention one thing that the SNR threshold should be lower that E4M3. We think the threshold set to 20 is ok. More details: https://github.com/AMD-AGI/Primus-Turbo/blob/c64bfd4e1e8c50943ec86caf81f1291b5980b4d2/tests/pytorch/ops/test_gemm_fp8.py#L97
